internal/provider/feature_resource.go

```diff
@@ -447,6 +428,50 @@ func (r *FeatureResource) step(feat *features.Feature, h harness.Harness, data F
 	return nil
 }
 
+func (r *FeatureResource) teardown(ctx context.Context, data FeatureResourceModel, h harness.Harness) diag.Diagnostics {
+	remaining, err := r.store.Inventory(data.Harness.Inventory).RemoveFeature(ctx, inventory.Feature{
+		Id:      data.Id.ValueString(),
+		Harness: inventory.Harness(data.Harness.Id.ValueString()),
+	})
+	if err != nil {
+		return []diag.Diagnostic{diag.NewErrorDiagnostic("failed to remove feature from inventory", err.Error())}
+	}
+
+	if len(remaining) == 0 {
+		log.Debug(ctx, "no more features remain in inventory, removing harness")
+		if err := r.store.Inventory(data.Harness.Inventory).RemoveHarness(ctx, inventory.Harness(data.Harness.Id.ValueString())); err != nil {
+			return []diag.Diagnostic{diag.NewErrorDiagnostic("failed to remove harness from inventory", err.Error())}
+		}
+
+		// Destroy the harness...
+		if r.store.SkipTeardown() {
+			return []diag.Diagnostic{
+				diag.NewWarningDiagnostic(
+					fmt.Sprintf("teardown for harness [%s] is skipped because IMAGETEST_SKIP_TEARDOWN is set", data.Harness.Id.ValueString()),
+					fmt.Sprintf(`There are dangling resources that will require manual cleanup.
+
+To remove the resources specific to this harness, run the following:
+
+  docker rm -f $(docker ps -a -q --filter "name=^%[1]s*" --filter "label=dev.chainguard.imagetest=true")
+  docker network rm -f $(docker network ls -q --filter "name=^%[1]s*" --filter "label=dev.chainguard.imagetest=true")
+
+To cleanup all resources owned by imagetest, run the following:
+
+  docker rm -f $(docker ps -a -q --filter "label=dev.chainguard.imagetest=true")
+  docker system prune --volumes --all
```

Comment:
these are hardcoded for our existing docker based harnesses and don't apply for `pterraform`.

ideally we plumbed the `TeardownString` or something through from the harness, but I don't want to overcomplicate things yet, we're not using `pterraform` anywhere yet